### PR TITLE
node-jose: added iv option to JWE.createEncrypt

### PR DESCRIPTION
**Pull request by ymkjp**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/cisco/node-jose/blob/1f86e23fc6/lib/jwe/encrypt.js#L666
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/node-jose/index.d.ts [hunk 1]
```diff
@@ -97,6 +97,7 @@ export namespace JWE {
         fields?: object;
         contentAlg?: string;
         protect?: string | string[];
+        iv?: Buffer;
```
Comment (peterblazejewicz) [inline types/node-jose/index.d.ts:100]: Can this be a `string`? As per source you've linked:
https://github.com/cisco/node-jose/blob/1f86e23fc6/lib/jwe/encrypt.js#L391-L395

Comment (ymkjp) [reply types/node-jose/index.d.ts]: Thanks for the review. With https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53788/commits/d2159fb5453a5c1d02f836f14cfc900985cad77f, I've made the following changes:

* added `string` and a test case for it
* clarified the use case for [`Buffer`](https://nodejs.org/api/crypto.html#crypto_crypto_randombytes_size_callback) in the test case

File: types/node-jose/index.d.ts [hunk 2]
```diff
@@ -97,6 +97,7 @@ export namespace JWE {
         fields?: object;
         contentAlg?: string;
         protect?: string | string[];
+        iv?: string|Buffer;
```
Comment (peterblazejewicz) [inline types/node-jose/index.d.ts:100]:
```suggestion
        iv?: string | Buffer;
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@ymkjp thx!

**Review by peterblazejewicz — APPROVED** [review]

**Comment by ymkjp** [comment]
Ready to merge

**Pull request merged**